### ISSUE TITLE
Option for running as service in background after boot.

### ISSUE THREAD

**Issue by dreaddr** [lawl/NoiseTorch#193]
First of all, amazing project... this has helped me tons since the fan noise of my device was driving me crazy and so I truly have to say thank you for making this great tool.

Okay, with that being said I noticed that while there is a way to run this program as a service detailed in the [Wiki](https://github.com/lawl/NoiseTorch/wiki/Start-automatically-with-Systemd), this is a very manual process and I'd just like to suggest having an option within the program itself to do this process automatically. (Maybe with a warning that CPU usage will be affected as long as it keeps running if thats a concern? NoiseTorch was using about 10% of my CPU at peak, which is a measurable amount but not noticeable on my personal day to day usage.)

Thanks for your work! I wish this gets added as it would be easy to recommend this tool to my (quite limited but still) known Linux users.

**Comment by lawl** [comment]
> NoiseTorch was using about 10% of my CPU at peak, which is a measurable amount but not noticeable on my personal day to day usage.

Yep, i know this is fine for a lot of people and systems, it's not though for laptops for example, as it will kill your battery life.

This is not a problem on pipewire system, where this doesn't cause CPU usage while not in use.

But pipewire has some other issues, and i'd prefer to keep the functionality between pipewire and pulseaudio consistent, otherwise it's confusing. A user shouldn't have to know what their audio stack is.

I've experimented with making things work on pulseaudio, see e.g. the "Microphone is in use" warning when you try to unload while it's in use. Unforunatelly, it doesn't work correctly and will still report "in use" even after you've stopped using in.

In short, I haven't found a way to do this with what I'd consider acceptable user experience yet. It's not that I don't *want* to do it.

**Comment by dreaddr** [comment]
Ah yes, I hadn't considered battery life... yeah it makes more sense now. 

I do have PipeWire installed but I believe NoiseTorch still uses PulseAudio because I'm still getting those usage numbers? Understandable that we don't have to consider the audio stack, most people are still on Pulse to this day unfortunately. Maybe one day you can safely switch the stacks.

So I can expect this to be considered and added down the line in the project as it matures then? You can close the issue if so, I hope it gets added sooner than later :smile:

**Comment by lawl** [comment]
> So I can expect this to be considered and added down the line in the project as it matures then

I don't know, you can expect it to be added once I figure out a good way to do it. That may be tomorrow, or never. I'll close this issue in the meantime, because there's unfortunately nothing that can be acted on right now.
